src/service/engine/v1/index.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  ACCOUNT_ENGINE,
+  ACCOUNT_ENGINES,
+  ACCOUNT_DATABASE_BINDING_URL,
+  ResultsPage
+} from "../../../common/api";
+import { ResourceManagerContext } from "../../../types";
+import { EngineModel } from "./model";
+import { Engine, ID } from "./types";
+import { CreateEngineOptions, WarmupMethod } from "../types";
+import { DatabaseModel } from "../../database/v1/model";
+import {
+  getCheepestInstance,
+  resolveEngineSpec,
+  resolveRegionKey
+} from "../../utils";
+import { ResourceManager } from "../../index";
+import { DatabaseService } from "../../database/v1";
+
+export class EngineService {
+  private readonly context: ResourceManagerContext;
+
+  constructor(context: ResourceManagerContext) {
+    this.context = context;
+  }
+
+  private get accountId(): Promise<string> {
+    return this.context.connection.resolveAccountId();
+  }
+
+  private async getEngineId(engineName: string): Promise<ID> {
+    const { apiEndpoint, httpClient } = this.context;
+    const queryParams = new URLSearchParams({ engine_name: engineName });
+    const url = `${apiEndpoint}/${ACCOUNT_ENGINES(
+      await this.accountId
+    )}:getIdByName?${queryParams}`;
+    const data = await httpClient
+      .request<{ engine_id: ID }>("GET", url)
+      .ready();
+    return data.engine_id;
+  }
+
+  async getById(engineId: string): Promise<EngineModel> {
+    const { apiEndpoint, httpClient } = this.context;
+    const url = `${apiEndpoint}/${ACCOUNT_ENGINE(
+      await this.accountId,
+      engineId
+    )}`;
+    const data = await httpClient
+      .request<{ engine: Engine }>("GET", url)
+      .ready();
+    return new EngineModel(this.context, data.engine);
+  }
+
+  async getByDB(database_name: string): Promise<EngineModel[]> {
+    return [];
+  }
```

Comment:
Is this not supported? Let's drop a comment if so.

Reply:
Missed this one, will add the implementation. Thanks for pointing